File: src/main/java/com/github/yuppieflu/stats/rest/dto/Statistic.java
package com.github.yuppieflu.stats.rest.dto;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class Statistic {
    private final double sum;
    private final double avg;
    private final double max;
    private final double min;
    private final long count;

    public static Statistic from(com.github.yuppieflu.stats.service.domain.Statistic statistic) {
        return Statistic.builder()
                        .count(statistic.getCount())
                        .max(statistic.getMax())
                        .min(statistic.getMin())
                        .avg(statistic.getAvg())
                        .sum(statistic.getSum())
                        .build();
    }
}

File: src/main/java/com/github/yuppieflu/stats/rest/dto/Statistic.kt
package com.github.yuppieflu.stats.rest.dto

import com.github.yuppieflu.stats.service.domain.Stat

class Statistic(
        override val count: Long,
        override val min: Double,
        override val max: Double,
        override val sum: Double
) : Stat {
    constructor(stat: Stat) : this(count = stat.count, min = stat.min, max = stat.max, sum = stat.sum)
}